# Desafio World Cup 2014

## Desafio 1

In [41]:
import pandas as pd
import numpy as np
df=pd.read_csv("worldcup2014.csv")
df.head(5)

team     continent group  group_pos  cantidad_juegos  \
0       Brazil  southamerica     A          1                3   
1       Mexico  northamerica     A          2                3   
2      Croatia        europe     A          3                3   
3     Cameroon        africa     A          4                3   
4  Netherlands        europe     B          1                3   

   juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0               2                 1                0            7   
1               2                 1                0            4   
2               1                 0                2            6   
3               0                 0                3            1   
4               3                 0                0           10   

   goles_contra goles_diferencia  puntos  clasificado  
0             2                5       7            1  
1             1                3       7            1  
2             6                0       3            0  
3             9               −8       0            0  
4             3                7       9            1

## Desafio 2

In [42]:
df.continent.value_counts()

europe          13
southamerica     6
africa           5
northamerica     4
asia             4
Name: continent, dtype: int64

In [43]:
df.continent.value_counts('%')

europe          0.40625
southamerica    0.18750
africa          0.15625
northamerica    0.12500
asia            0.12500
Name: continent, dtype: float64

### ¿Cuál es el continente con una mayor presencia en la muestra?
Europe
### ¿Cuál es la probabilidad de elegir un equipo asiático al azar?
12,5%
### ¿Cuál es la probabilidad de elegir un equipo africano al azar?
15.625%

## Desafio 3

In [140]:
df_europe=df.loc[df.continent=='europe']
df_southamerica=df.loc[df.continent=='southamerica']
df_africa=df.loc[df.continent=='africa']
df_northamerica=df.loc[df.continent=='northamerica']
df_asia=df.loc[df.continent=='asia']

In [145]:
df_europe.clasificado.value_counts('%')

0    0.538462
1    0.461538
Name: clasificado, dtype: float64

In [146]:
df_southamerica.clasificado.value_counts('%')

1    0.833333
0    0.166667
Name: clasificado, dtype: float64

In [147]:
df_africa.clasificado.value_counts('%')

0    0.6
1    0.4
Name: clasificado, dtype: float64

In [148]:
df_northamerica.clasificado.value_counts('%')

1    0.75
0    0.25
Name: clasificado, dtype: float64

In [149]:
df_asia.clasificado.value_counts('%')

0    1.0
Name: clasificado, dtype: float64

### ¿Cuál es la probabilidad de que un país asiático pase a la siguiente ronda?
La probabilidad de que un país asiatico pase a la siguente ronda es de un 0%

### ¿Cuáles son los dos continentes con la mayor probabilidad de clasificar?
Los continentes con mayor probabilidad de clasificar son:
- df_southamerica con un 83.3%
- df_northamerica con un 75%


### ¿Cuál es la probabilidad de que un país europeo no clasifique?
La probabilidad de que un país europeo no clasifique es de un 53,8462%

## Desafío 5: Probabilidades de Victoria y Derrota

In [10]:
df['ha_ganado']=0
df['ha_ganado']=np.where(df.juegos_ganados>0,1,0)
df[['team','juegos_ganados','ha_ganado']].head()

team  juegos_ganados  ha_ganado
0       Brazil               2          1
1       Mexico               2          1
2      Croatia               1          1
3     Cameroon               0          0
4  Netherlands               3          1

In [11]:
df_europe=df.loc[df.continent=='europe']
df_southamerica=df.loc[df.continent=='southamerica']
df_africa=df.loc[df.continent=='africa']
df_northamerica=df.loc[df.continent=='northamerica']
df_asia=df.loc[df.continent=='asia']

In [12]:
df_europe.ha_ganado.value_counts('%')

1    0.846154
0    0.153846
Name: ha_ganado, dtype: float64

In [13]:
df_southamerica.ha_ganado.value_counts('%')

1    1.0
Name: ha_ganado, dtype: float64

In [14]:
df_africa.ha_ganado.value_counts('%')

1    0.6
0    0.4
Name: ha_ganado, dtype: float64

In [15]:
df_northamerica.ha_ganado.value_counts('%')

1    0.75
0    0.25
Name: ha_ganado, dtype: float64

In [17]:
df_asia.ha_ganado.value_counts('%')

0    1.0
Name: ha_ganado, dtype: float64

### ¿Qué continente tiene la mayor probabilidad de tener países con al menos 1 victoria?

El continente que tiene la mayor probabilidad de tener países con al menos 1 victoria es southamerica con una probabilidad de un 100%

### ¿Qué continente tiene un nivel similar entre países que tienen o no tienen victorias?
El continente que tiene un nivel similar entre paises que tienen y no tienen victoria es africa, con una probabilidad de un 60% de que tenga a lo menos una victoria y un 40% sin ninguna victoria

### Reformule el código para juegos perdidos y analice los continentes con mayor probabilidad de tener países sin juegos perdidos

In [30]:
df['ha_perdido']=0
df['ha_perdido']=np.where(df.juegos_perdidos>0,1,0)
df[['team','juegos_perdidos','ha_perdido']].head()

team  juegos_perdidos  ha_perdido
0       Brazil                0           0
1       Mexico                0           0
2      Croatia                2           1
3     Cameroon                3           1
4  Netherlands                0           0

In [31]:
df.ha_perdido.value_counts('%')

1    0.71875
0    0.28125
Name: ha_perdido, dtype: float64

In [32]:
df_europe=df.loc[df.continent=='europe']
df_southamerica=df.loc[df.continent=='southamerica']
df_africa=df.loc[df.continent=='africa']
df_northamerica=df.loc[df.continent=='northamerica']
df_asia=df.loc[df.continent=='asia']

In [33]:
df_europe.ha_perdido.value_counts('%')

1    0.692308
0    0.307692
Name: ha_perdido, dtype: float64

In [34]:
df_southamerica.ha_perdido.value_counts('%')

0    0.5
1    0.5
Name: ha_perdido, dtype: float64

In [35]:
df_africa.ha_perdido.value_counts('%')

1    1.0
Name: ha_perdido, dtype: float64

In [36]:
df_northamerica.ha_perdido.value_counts('%')

0    0.5
1    0.5
Name: ha_perdido, dtype: float64

In [37]:
df_asia.ha_perdido.value_counts('%')

1    1.0
Name: ha_perdido, dtype: float64

### ¿Qué continente tiene la mayor probabilidad de tener países sin juegos perdidos?

Los continentes que tienen una probabilidad de un 50% de tener paises sin juegos perdidos son southamerica y northamerica
